dart-impl/mpi/src/dart_mem.c

```diff
@@ -166,7 +171,7 @@ dart_buddy_alloc(struct dart_buddy * self, size_t s) {
 		}
 		for (;;) {
 			level--;
-			length *= 2;
+//    length *= 2;
```

Comment:
This is strange: the second `if` below breaks the `for(;;;)` loop and thus the value _is_ read at the beginning of the enclosing `while` loop. Which Clang version do you use? I do not see this warning with 3.8.0.